core/trino-main/src/main/java/io/trino/sql/planner/optimizations/ReplicateJoinAndSemiJoinInDelete.java

```diff
@@ -51,10 +52,11 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
         {
             PlanNode leftSourceRewritten = context.rewrite(node.getLeft(), context.get());
 
+            // This should be applied to Joins directly between TableScan and DeleteNode, not to all Joins in the plan
             JoinNode rewrittenNode = (JoinNode) node.replaceChildren(ImmutableList.of(leftSourceRewritten, node.getRight()));
             Type joinType = rewrittenNode.getType();
             if (isDeleteQuery && (joinType == INNER || joinType == LEFT)) {
-                return rewrittenNode.withDistributionType(REPLICATED);
+                return rewrittenNode.withDistributionType(DistributionType.REPLICATED);
```

Comment:
is this `JoinNode.DistributionType` or `SemiJoinNode.DistributionType`? (both are used in this class)

qualify `JoinNode.DistributionType.REPLICATED`